config/prow/deploy.sh

```diff
@@ -85,6 +85,11 @@ kubectl apply --server-side=true -k "$SCRIPT_DIR/cluster/ingress-nginx"
 kubectl wait --for=condition=available deployment -l app.kubernetes.io/component=controller -n ingress-nginx --timeout 2m
 echo "$(color-green done)"
 
+echo "$(color-step "Deploying oauth2-proxy components to gardener-prow-trusted cluster...")"
+kubectl config use-context gardener-prow-trusted
+kubectl apply --server-side=true -k "$SCRIPT_DIR/cluster/oauth2-proxy"
```

Comment:
Instead of deploying this in an extra step, you could also include oauth2-proxy directly in the `monitoring/trusted-cluster` kustomization :)